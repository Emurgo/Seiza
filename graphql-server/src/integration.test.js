import {createTestClient} from 'apollo-server-testing'
import gql from 'graphql-tag'
import createServer from './server'

// eslint-disable-next-line no-console
console.log = () => {
  /* be silent */
}

// We need to override this before loading the server
process.env.APOLLO_ENGINE_API_KEY = null
const server = createServer()
const {query} = createTestClient(server)

jest.setTimeout(30000)

const ADDRESS_QUERY = gql`
  query($address58: String!) {
    address(address58: $address58) {
      address58
      transactionsCount
      balance
      totalAdaSent
      totalAdaReceived
    }
  }
`

it('Gets correct results for huge address', async () => {
  const address58 =
    'DdzFFzCqrhsfkegDcdUJAGBRoUP2LVakkby6ntdckcURzBsKmNJ7HmQ6LBwLZxTRVBvhZzuFuX9KUpraDcqhJavm35yeXgS2keJPHfKB'
  const res = await query({
    query: ADDRESS_QUERY,
    variables: {
      address58,
    },
  })

  expect(res).toMatchSnapshot()
})
