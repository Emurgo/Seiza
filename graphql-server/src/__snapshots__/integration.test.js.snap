// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gets correct results for huge address 1`] = `
Object {
  "data": Object {
    "address": Object {
      "address58": "DdzFFzCqrhsfkegDcdUJAGBRoUP2LVakkby6ntdckcURzBsKmNJ7HmQ6LBwLZxTRVBvhZzuFuX9KUpraDcqhJavm35yeXgS2keJPHfKB",
      "balance": "27275000000",
      "totalAdaReceived": "10041946861122765",
      "totalAdaSent": "10041919586122765",
      "transactionsCount": 102757,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=20, public",
        ],
      },
    },
  },
}
`;
